apps/communications/contacts/js/views/selector.js

```diff
@@ -0,0 +1,131 @@
+
+Contacts.Selector = (function(){
+  var selectorHeader = document.getElementById('selector-view-header');
+  var saveButton = document.getElementById('save-button');
+  var container = document.getElementById('selector-container');
+  var template = document.getElementById('contactToImportTemplate');
+
+  saveButton.addEventListener('click', importAll);
+  selectorHeader.addEventListener('action', cancelHandler);
+  container.addEventListener('click', clickHandler);
+
+  var contactsToImport = [];
+
+  function cancelHandler() {
+    selectorHeader.removeEventListener('action', cancelHandler);
+    saveButton.removeEventListener('click', importAll);
+    container.removeEventListener('click', clickHandler);
+    Contacts.cancel();
+  }
+
+  function clickHandler(event) {
+    // FIXME Seems to be a conflict when calling an activity from an activity.
+    /*var id = event.target.dataset.tempId;
+    ActivityHandler.mozContactParam = contactsToImport[id];
+    var activity = {
+      source: {
+        name: 'new',
+        data: {
+          type: ['text/vcard'],
+          allowSave: true,
+          params: {
+            mozContactParam: true
+          }
+        }
+      }
+    };
+    ActivityHandler.handle(activity);*/
+  }
+
+  function importAll() {
+    var numDupsMerged = 0,
+        importedContacts = 0;
+
+    contactsToImport.forEach((contact, index) => {
+      contacts.Matcher.match(contact, 'passive', {
+        onmatch: matches => {
+          var callbacks = {
+            success: function(mergedContact) {
+              numDupsMerged++;
+              _doContinue(index);
+            },
+            error: _doContinue.bind(this, index)
+          };
+          contacts.adaptAndMerge(contact, matches, callbacks);
+        },
+        onmismatch: function() {
+          var saving = navigator.mozContacts.save(contact);
+
+          saving.onsuccess = () => {
+            _doContinue(index, true);
+          };
+
+          saving.onerror = (err) => {
+            console.error(err);
+            _doContinue(index);
+          };
+        }
+      });
+    });
+
+    function _doContinue(index, isContactImported) {
+      if (isContactImported) {
+        importedContacts++;
+      }
+
+      if (contactsToImport.length - 1 <= index) {
+        utils.status.show({
+          id: 'memoryCardContacts-imported3', // FIXME show better message.
+          args: {n: importedContacts}
+        }, {
+          id: 'contactsMerged',
+          args: {numDups: numDupsMerged}
+        });
+
+        window.setTimeout(cancelHandler, 1500); // Allows to see status message.
+      }
+    }
+  }
+
+  /**
+   * Allows to preview which contacts will be imported.
+   * @param contacts - Every contact already read from vCard.
+   * @param cursor - Reader object to continue reading from vCard.
+   */
+  var addContacts = (contacts, cursor) => {
+    contacts.forEach(_doAddContact);
+
+    cursor.onsuccess = event => {
```

Comment:
nit: always uses parenthesis between parameters in arrow functions, it makes the code more readable